File: d56c154c-697a-4e54-90a1-12cac4adf267/d56c154c-697a-4e54-90a1-12cac4adf267.c
/*
NAME: d56c154c-697a-4e54-90a1-12cac4adf267.c
QUESTION: Do I have root privileges?
CREATED: 2022-10-28 10:08:27.169993
*/
#include <stdlib.h>
#include <string.h>
#include <stdio.h>

#define BUFSIZE 100

void test(void)
{
    char user[BUFSIZE];
    char *envvar = "USER";

    if (!getenv(envvar))
    {
        exit(1);
    }

    if (snprintf(user, BUFSIZE, "%s", getenv(envvar)) >= BUFSIZE)
    {
        exit(1);
    }

    if (strcmp(user, "root") == 0)
    {
        exit(100);
    }
    else
    {
        exit(101);
    }
}

void clean(void)
{
    exit(100);
}

int main(int argc, char *argv[])
{
    if (argc > 1)
    {
        clean();
    }
    else
    {
        test();
    }
}

File: variants/d56c154c-697a-4e54-90a1-12cac4adf267/d56c154c-697a-4e54-90a1-12cac4adf267.go
/*
NAME: d56c154c-697a-4e54-90a1-12cac4adf267.c
QUESTION: Is there evidence of LockBit ransomware?
CREATED: 2022-12-30
*/
package main

import (
	"github.com/preludeorg/test/basic"
	"os"
)

func test() {
	println("[*] Running test")
	evidence := basic.Find(".lockbit")
	if evidence != nil {
		println("[!] Discovered evidence of LockBit ransomware")
		os.Exit(101)
	}
	os.Exit(100)
}

func clean() {
	println("[*] Running clean")
	os.Exit(100)
}

func main() {
	args := os.Args[1:]
	if len(args) > 0 {
		clean()
	} else {
		test()
	}
}
